taipy/gui/builder/_element.py

```diff
@@ -11,18 +11,55 @@
 
 from __future__ import annotations
 
+import ast
+import builtins
 import copy
+import inspect
+import io
 import re
+import sys
 import typing as t
 from abc import ABC, abstractmethod
 from collections.abc import Iterable
+from operator import attrgetter
+from types import FrameType, FunctionType
 
+from .._warnings import _warn
+
+if sys.version_info < (3, 9):
+    from ..utils.unparse import _Unparser
 from ._context_manager import _BuilderContextManager
 from ._factory import _BuilderFactory
 
 if t.TYPE_CHECKING:
     from ..gui import Gui
 
+python_builtins = dir(builtins)
+
+
+def _get_value_in_frame(frame: FrameType, name: str):
+    if not frame:
+        return None
+    if name in frame.f_locals:
+        return frame.f_locals.get(name)
+    return _get_value_in_frame(t.cast(FrameType, frame.f_back), name)
+
+
+class _TransformVarToValue(ast.NodeTransformer):
+    def __init__(self, frame: FrameType, non_vars: t.List[str]) -> None:
+        super().__init__()
+        self.frame = frame
+        self.non_vars = non_vars
+
+    def visit_Name(self, node):
+        var_name = node.id.split(".", 2)[0]
+        if var_name in self.non_vars:
+            return node
+        value = _get_value_in_frame(self.frame, var_name)
```

Comment:
This line and line 60 should be swapped by changing the if condition since this is more expensive to run recursively